common/buildcraft/lib/marker/MarkerSubCache.java

```diff
@@ -103,6 +103,15 @@ public void removeMarker(BlockPos pos) {
         }
     }
 
+    public void removeConnection(BlockPos pos1, BlockPos pos2) {
+        for(MarkerConnection connection : getConnections()) {
+            BlockPos[] positions = (BlockPos[]) connection.getMarkerPositions().toArray(new BlockPos[0]);
+            if(Arrays.binarySearch(positions, pos1) >= 0 && Arrays.binarySearch(positions, pos2) >= 0) {
+                deinitConnection(new HashSet<>(Arrays.asList(positions)));
```

Comment:
A few things:

1. Removing connections looks odd, and is generally easier to destroy the marker itself rather than the connection (as suddenly there are a lot of connection paths that looks odd). Basically "remove this please".

2. You can't perform a binary search on an unsorted array/list.

3. Just blindly using ````deinitConnection```` doesn't work.